library/core/crash-info/src/main/java/org/quiltmc/qsl/crash/mixin/WorldMixin.java

```diff
@@ -0,0 +1,19 @@
+package org.quiltmc.qsl.crash.mixin;
+
+import net.minecraft.util.crash.CrashReport;
+import net.minecraft.util.crash.CrashReportSection;
+import net.minecraft.world.World;
+import org.quiltmc.qsl.crash.api.CrashReportEvents;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(World.class)
+public class WorldMixin {
+	@Inject(method = "addDetailsToCrashReport", at = @At("TAIL"), locals = LocalCapture.CAPTURE_FAILSOFT)
```

Comment:
Wasn't sure about `LocalCapture.CAPTURE_FAILSOFT` - do we prefer crashes for this?

Reply:
I believe we want to do fail hard everywhere

Reply:
I recall some odd behavior in a past mc version where this may have been initialized at some point earlier in game lifecycle?

Reply:
If that behavior still applies, the javadoc should make it clear this is just for obtaining info and a crash may not be in progress?

Reply:
Looks like it's also used to create a debug report (from /debug)